File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/Java/c92c27565ddf4f5c66fa315e1c48ce9d.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int k = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = -sc.nextInt();
        }
        Arrays.sort(a);
        int ans = 0;
        int res = 0;
        for (int i = 0; i < n; i++) {
            res -= a[i];
            if (res < 0) {
                ans += res;
                break;
            }
            if (i != n - 1) {
                ans += res;
            }
        }
        if (ans >= 0) {
            System.out.println(ans);
            return;
        }
        List<Integer> tmp = new ArrayList<>();
        tmp.add(res);
        for (int i = j + 1; i < n; i++) {
            tmp.add(-a[i]);
        }
        Collections.sort(tmp);
        int w = 0;
        for (int c : tmp) {
            ans += (w / (k + 1)) * c;
            w++;
        }
        System.out.println(ans);
    }
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Go/c92c27565ddf4f5c66fa315e1c48ce9d.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var n, k int
	fmt.Scan(&n, &k)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
		a[i] = -a[i]
	}
	sort.Ints(a)
	ans := 0
	res := 0
	j := -1
	for i := 0; i < n; i++ {
		res -= a[i]
		if res < 0 {
			j = i
			break
		}
		if i != n-1 {
			ans += res
		}
	}
	if j == -1 {
		fmt.Println(ans)
		return
	}
	tmp := make([]int, 0)
	tmp = append(tmp, res)
	for i := j + 1; i < n; i++ {
		tmp = append(tmp, -a[i])
	}
	sort.Ints(tmp)
	w := 0
	for _, c := range tmp {
		ans += (w / (k + 1)) * c
		w++
	}
	fmt.Println(ans)
}
